lib/SILGen/SILGenPoly.cpp

```diff
@@ -3714,7 +3714,7 @@ SILGenModule::getOrCreateAutoDiffDerivativeReabstractionThunk(
   // This thunk is publicly exposed and cannot be transparent.
   // Instead, mark it as "always inline" for optimization.
   auto *thunk = fb.getOrCreateFunction(
-      loc, name, SILLinkage::Hidden, origDerivativeFnType, IsBare,
+      loc, name, original->getLinkage(), origDerivativeFnType, IsBare,
```

Comment:
The linkage should be the same as the differentiability witness, not the original function.

Reply:
It was already using the original function linkage before this PR, so I will leave it as it is for now.

There are a big number of problems caused by incorrect linkage piling up, so I do want to carefully think through the correct linkage for everything and create a PR that fixes all the linkage soon. I have filed a ticket for that: https://bugs.swift.org/browse/TF-1027